File: LeetCode75_LevelTwo/Number_of_Provinces/Number_of_Provinces.py
# 547. Number of Provinces Python

class Solution(object):
    def findCircleNum(self, isConnected):
        """
        :type isConnected: List[List[int]]
        :rtype: int
        """
        length = len(isConnected)
        visited = [0] * length
        count = 0
        for i in range(length):
            if visited[i] == 0 :
                self.dfs(isConnected, visited, i)
                count+=1
        return count
    def dfs(self, isConnected, visited, i):
        for j in range(len(isConnected)):
            if isConnected[i][j] == 1 and visited[j] == 0:
                visited[j] = 1
                self.dfs(isConnected, visited, j)

File: LeetCode75_LevelTwo/Number_of_Provinces/Number_of_Provinces.cc
// 547. Number of Provinces C++
#include <vector>
using namespace std;

class Solution {
    int len;
public:
    int findCircleNum(vector<vector<int>>& isConnected) {
        len = isConnected.size();
        int count = 0;
        vector<int> visited(len, 0);
        for (int i = 0; i < len; i++) {
            if (visited[i] ==0){
                dfs(isConnected, visited, i);
                count++;
            }
        }
        return count;
    }
    void dfs(vector<vector<int>>& isConnected, vector<int>& visited, int i) {
        for (int j = 0; j < len; j++) {
            if (isConnected[i][j] == 1 && visited[j] == 0) {
                visited[j] = 1;
                dfs(isConnected, visited, j);
            }
        }
    }
};
